### DISCUSSION TITLE
Seeking Solutions for Particle Escape Error in High Thermal Fluctuation Simulations with Barriers

### DISCUSSION THREAD

**Discussion by idop11** [glotzerlab/hoomd-blue#1761] [Support] [answered]
Hello,

I'm encountering an issue when simulating a system of Brownian particles subjected only to thermal fluctuations, with barriers (I've encountered this in the case of both Wall potential and Fixed particles; two scripts are attached for reference).
Whenever I attempt to simulate the system with a large value of $kT$ (specifically for $kT>3$) with $dt=0.01$, I encounter an error as follows (this occurs with both types of barriers):

```
Traceback (most recent call last):
  File "/mnt/c/Users/User/PycharmProjects/graphicsExperiments/sample_code_particleBarrier.py", line 104, in <module>
    simulation.run(N_steps)
  File "/home/idop/miniconda3/envs/hoomd_newest/lib/python3.11/site-packages/hoomd/simulation.py", line 562, in run
    self._cpp_sys.run(steps_int, write_at_start)
RuntimeError: Particle with unique tag 387 is no longer in the simulation box.
```

I understand this issue is related to setting too large a value of $dt$ for these specific values of $kT$. However, I need to maintain this $dt$ value for consistency with previous simulations. I'm seeking advice on how I might explore higher values of $kT$ with $dt=0.01$. Would using ```r_extrap``` overcome this issue, for example?

The two scripts are:
[sample_code_Wall.txt](https://github.com/glotzerlab/hoomd-blue/files/14905903/sample_code_Wall.txt)
[sample_code_particleBarrier.txt](https://github.com/glotzerlab/hoomd-blue/files/14905904/sample_code_particleBarrier.txt)

Thanks,
Ido

**Comment by joaander** [comment, marked as answer]
I'm surprised that your simulations ran successfully before. In my experience, dt=0.01 is far too large to be stable for WCA simulations with the `ConstantVolume` and `ConstantPressure` integration methods. I have often used dt=0.005 in the past for LJ/WCA simulations (epsilon=1, sigma=1, kT=~1-3). Although [detailed testing](https://github.com/glotzerlab/hoomd-validation/) shows that dt=0.001 is necessary for accurate results.

You are also using `Brownian` which requires even smaller dt due to its O(dt) error (compared to O(dt**2) for `ConstantVolume` and `ConstantPressure`).

**Reply by idop11** [reply]
Thank you, Joshua.

Unfortunately, changing dt=0.01 will complicate other aspects of my simulations, so I'll likely need to stick with it. However, switching to Langevin dynamics does indeed allow for the examination of higher kT values. Do you have any other suggestions for scenarios where I must stick with Brownian dynamics?

**Reply by joaander** [reply]
No.
